engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/preparation/CallPreparation.scala

```diff
@@ -23,23 +28,73 @@ object CallPreparation {
-      expressionLanguageFunctions,
-      outputStore.fetchNodeOutputEntries,
-      scatterMap
-    ) recoverWith {
-      case t: Throwable => Failure[Map[Declaration, WdlValue]](new VariableLookupException(s"Couldn't resolve all inputs for ${callKey.scope.fullyQualifiedName} at index ${callKey.index}.", List(t)))
+
+    val inputMappingsFromPreviousCalls: Map[String, WdlValue] = {
+      call.inputPorts.flatMap({ inputPort =>
+        val outputPort = inputPort.upstream
+        // TODO WOM: scatters ?
+        outputStore.get(outputPort, None) map { outputPort.fullyQualifiedName -> _ }
+      }) toMap
+    }
+
+    // Previous calls outputs and known workflow level values
+    val externalInputMappings = inputMappingsFromPreviousCalls ++ workflowDescriptor.backendDescriptor.knownValues
+
+    def resolveInputDefinitionFold(accumulatedInputsSoFar: Map[InputDefinition, ErrorOr[WdlValue]],
+                                   inputDefinition: InputDefinition): Map[InputDefinition, ErrorOr[WdlValue]] = {
+
+      val validInputsAccumulated: Map[String, WdlValue] = accumulatedInputsSoFar.collect({
+        case (input, Valid(errorOrWdlValue)) => input.name -> errorOrWdlValue
+      }) ++ externalInputMappings
+
+      // Tries to evaluate a WomExpression and coerce it to the desired type
+      def evaluateAndCoerce(expression: WomExpression, coerceTo: WdlType): ErrorOr[WdlValue] = {
+        expression.evaluateValue(validInputsAccumulated, expressionLanguageFunctions) flatMap {
+          coerceTo.coerceRawValue(_).toErrorOr
+        }
+      }
+
+      // Tries to find the current input definition in the map of known values
+      def resolveFromKnownValues: ErrorOr[WdlValue] = {
+        validInputsAccumulated.get(inputDefinition.name)
+          // FullyQualified names are not working so make one up ourselves. Again this will be unnecessary when wom give us mappings
+          .orElse(validInputsAccumulated.get(workflowDescriptor.workflow.name + "." + call.unqualifiedName + "." + inputDefinition.name)) match {
+          case Some(value) => value.validNel
+          case None => s"Can't find a known value for ${inputDefinition.name}".invalidNel
+        }
+      }
+
+      // Tries to find the current input definition in the list of expression based inputs, and if so evaluates it
+      def resolveAsExpression: ErrorOr[WdlValue] = {
+        call.expressionBasedInputs.get(inputDefinition.name) map { instantiatedExpression =>
+          evaluateAndCoerce(instantiatedExpression.expression, inputDefinition.womType)
+        } getOrElse s"Can't find an expression to evaluate for ${inputDefinition.name}".invalidNel
+      }
+
+      // Tries to resolve the input definition by first looking in the known values, otherwise evaluate the associated
+      // expression, if any. The order is important because we want to prioritize a value passed through workflow input over
+      // an expression. This effectively allows to "override" the expression.
```

Comment:
@cjllanwarne's favorite WDL feature. I think this also covers explicit overrides in a call, whether it was a workflow input or not?